### PR TITLE
[org] Add org-appear support

### PR DESCRIPTION
**Pull request by winsphinx**
One picture is more than many words :  

image

Update PR at #14482

### CODE REVIEW

**Review by Kazark — APPROVED** [review]

File: CHANGELOG.develop [hunk 1]
```diff
@@ -803,6 +803,7 @@ Other:
     (thanks to rayw000)
   - Added =string-edit= package to =spacemacs-editing= layer
     (thanks to Ivan Yonchovski)
+  - Added =org-appear= package to =spacemacs-org= layer (thanks to winsphinX)
```
Comment (Kazark) [inline CHANGELOG.develop:806]: Should this be under the Org layer instead of in "Distribution changes"?

**Review by Kazark — COMMENTED** [review]
I have patched my own fork with these changes and they look good. Thanks for this; I had long wanted a feature like this, but didn't know about `org-appear`.

@smile13241324 could we look at getting this merged? I think this is a wonderful enhancement.

File: CHANGELOG.develop (hunk 1 above)
Comment (smile13241324) [reply CHANGELOG.develop:806]: Yes indeed, please move this comment to the org layer section.

File: layers/+emacs/org/README.org [hunk 2]
```diff
@@ -413,6 +414,14 @@ Set the variable =org-enable-valign= to =t=:
 
 [[https://github.com/casouri/valign#valignel][Known problem: Rendering large tables (≥100 lines) is laggy.]]
 
+** Org-appear support
+To install [[https://github.com/awth13/org-appear][org-appear]], set the =org-enable-appear-support= to =t=:
```
Comment (smile13241324) [inline layers/+emacs/org/README.org:418]: Can we have a high-level description about what org-appear does here? In the end we could refer to the original repository for more details.

File: layers/+emacs/org/packages.el [hunk 3]
```diff
@@ -50,7 +50,9 @@
     (org-sticky-header :toggle org-enable-sticky-header)
     (verb :toggle org-enable-verb-support)
     (org-roam :toggle org-enable-roam-support)
-    (valign :toggle org-enable-valign)))
+    (valign :toggle org-enable-valign)
+    (org-appear :location(recipe :fetcher github :repo "awth13/org-appear")
```
Comment (smile13241324) [inline layers/+emacs/org/packages.el:54]: Missing blank between location and bracket.

**Review by smile13241324 — CHANGES_REQUESTED** [review]
I have just reviewed your PR @winsphinx, I like the new feature however I still think we should tweak it a bit before we merge it though.

Please have a look at my comments, if you have questions or need help just drop me a line.

**Comment by winsphinx** [comment]
Thanks for your great help! @smile13241324 
I have just tried to fix them accoring your detail comments, and I hope you would review them, and please let me know if where is any mistake more.

Thank you again.

**Comment by smile13241324** [comment]
Your new PR looks good, however next time please just update your existing PR. I'll close this one in favor of #14482

**Comment by winsphinx** [comment]
Thanks.
It's a mistake I deleted the fork repo, and now I tried to pull PR but it can't be linked and combined when pushed the new branch.

**Pull request closed without merging**